pkgs/development/python-modules/functorch/default.nix

```diff
@@ -0,0 +1,40 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+, ninja
+, pytorch
+, which
+}:
+
+buildPythonPackage rec {
+  pname = "functorch";
+  version = "0.1.1";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-FidM04Q3hkGEDr4dthJv0MWtGiRfnWxJoyzu7Wl3SD8=";
+  };
+
+  # Somewhat surprisingly pytorch is actually necessary for the build process.
+  # `setup.py` imports `torch.utils.cpp_extension`.
+  nativeBuildInputs = [
+    ninja
+    pytorch
+    which
+  ];
+
+  # Tests require expecttest which is not yet packaged as of 2022-05-24.
```

Comment:
Might be worth packaging. Unfortunately, it comes up rather regularly, I even keep an [expression](https://github.com/SomeoneSerge/pkgs/blob/master/pkgs/ezy-expecttest.nix) for it

Reply:
That looks rather easy to package.

Reply:
Done in https://github.com/NixOS/nixpkgs/pull/174248/commits/b63b1f2266b7d748002e97566bf45b8f5f17d51f